### ISSUE TITLE
False negative on the `Style/OptionalBooleanParameter` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12401]
https://rubystyle.guide/#keyword-arguments-vs-option-hashes

https://rubystyle.guide/#boolean-keyword-arguments

## Describe the solution you'd like

```ruby
class Test
  # bad - common hack before keyword args were introduced
  def some_method(options = {})
    bar = options.fetch(:bar, false)
    puts bar
  end
end
```

Terminal:

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Rubocop

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.57.2 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by koic** [comment]
This issue was implemented in `StyleOptionHash` cop, which is disabled by default.
